training/components.py

```diff
@@ -296,156 +231,194 @@ def list_phase1_final_model():
+        name="NPROC_PER_NODE", value=f"{nproc_per_node}"
+    )
+    xdg_cache_var = models.V1EnvVar(name="XDG_CACHE_HOME", value="/tmp")
+    triton_cache_var = models.V1EnvVar(name="TRITON_CACHE_DIR", value="/tmp")
+    hf_home_var = models.V1EnvVar(name="HF_HOME", value="/tmp")
+    transformers_cache_var = models.V1EnvVar(name="TRANSFORMERS_CACHE", value="/tmp")
+
+    # Get master and worker container specs
+    master_container_spec = utils.get_container_spec(
+        base_image=base_image,
+        name="pytorch",
+        resources=resources_per_worker,
+        volume_mounts=[
+            input_data_volume_mount,
+            input_model_volume_mount,
+            output_volume_mount_master,
+        ],
+    )
+
+    # In the next release of kubeflow-training, the command
+    # and the args will be a part of utils.get_container_spec function
+    master_container_spec.command = command
+    master_container_spec.args = master_args
+
+    master_container_spec.env = [
+        nnodes_var,
+        nproc_per_node_var,
+        xdg_cache_var,
+        triton_cache_var,
+        hf_home_var,
+        transformers_cache_var,
+    ]
+
+    worker_container_spec = utils.get_container_spec(
+        base_image=base_image,
+        name="pytorch",
+        resources=resources_per_worker,
+        volume_mounts=[
+            input_data_volume_mount,
+            input_model_volume_mount,
+            output_volume_mount_worker,
+        ],
+    )
+    worker_container_spec.command = command
+    worker_container_spec.args = worker_args
+    worker_container_spec.env = [
+        nnodes_var,
+        nproc_per_node_var,
+        xdg_cache_var,
+        triton_cache_var,
+        hf_home_var,
+        transformers_cache_var,
+    ]
+
+    # create master pod spec
+    master_pod_template_spec = utils.get_pod_template_spec(
+        containers=[master_container_spec],
+        volumes=[input_data_volume, input_model_volume, output_volume],
+    )
+
+    # create worker pod spec
+    worker_pod_template_spec = utils.get_pod_template_spec(
+        containers=[worker_container_spec],
+        volumes=[input_data_volume, input_model_volume, output_volume],
+    )
+
+    logging.getLogger(__name__).setLevel(logging.INFO)
+    logging.info("Generating job template.")
+    logging.info("Creating TrainingClient.")
+
+    # Initialize training client
+    # This also finds the namespace from /var/run/secrets/kubernetes.io/serviceaccount/namespace
+    # And it also loads the kube config
+    training_client = TrainingClient()
+    namespace = training_client.namespace
+
+    # Create pytorch job spec
+    job_template = utils.get_pytorchjob_template(
+        name=name,
+        namespace=namespace,
+        worker_pod_template_spec=worker_pod_template_spec,
+        master_pod_template_spec=master_pod_template_spec,
+        num_workers=nnodes,
+        num_procs_per_worker=nproc_per_node,
+    )
+
+    print(job_template.to_str())
```

Comment:
Is it a debug or something you want to keep?

Reply:
I think it would be nice if we wrote this to disc to be inspected.

Reply:
Added as a yaml output artifact of the launcher component.